Src/FluentAssertions/CallerIdentification/AddNonEmptySymbolParsingStrategy.cs

```diff
@@ -16,7 +16,7 @@ public ParsingState Parse(char symbol, StringBuilder statement)
         }
         else if (mode is Mode.RemoveSuperfluousWhitespace)
         {
-            if (precedingSymbol is char value && !char.IsWhiteSpace(value))
+            if (precedingSymbol is { } value && !char.IsWhiteSpace(value))
```

Comment:
❓ I don't get this one. Why is it better?

Reply:
The general idea is that instead of checking if an object is an instance of a particular type, pattern matching let's you check that is not null and still assign it to a variable.
The benefit comes with inheritance where `is T` could check if the object is an instance of _another_ type, whereas `is { }` will just check if "is something", i.e. not null.

When the compiler can see that `is T` can _only_ be a null-check, i.e. the object is either null or an instance of `T`, `is T` also compiles down to a simple null check like `is {}` or `is not null`.

Reply:
as `is {}` is the same as `is not null`, would the last one not be the preferable notation?

Reply:
OK. That makes sense. A link to the relevant post would have been enough to convince me ;-)

Reply:
> as `is {}` is the same as `is not null`, would the last one not be the preferable notation?

`is not null` does not support declaration patterns, i.e. assigning the result to new variable, which is used here to check that `char? precedingSymbol` has a value _and_ extract that value to `char value`.

> OK. That makes sense. A link to the relevant post would have been enough to convince me ;-)

Why do you need blog posts when you have me?